### PR TITLE
added swagger docker

### PR DESCRIPTION
**Pull request by meltedspork**
This is first round with swagger docker - there are a bit of gotcha tho like no way to stream development port to yaml file

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -37,6 +37,14 @@ server.use(utils.auth.jwtCheck.unless({
   path: ['/', '/name', '/version']
 }))
 
+
+server.use(function crossOrigin(req,res,next){
```
Comment (kratsg) [inline src/index.js]: restify comes with cors support. Just use that instead of rolling your own.

**Review by kratsg — CHANGES_REQUESTED** [review]
Not 100% sure about this. It's not clear how we can test this specification against the API.

File: src/index.js (hunk 1 above)
Comment (meltedspork) [reply src/index.js]: where is that documented? I searched and coudnlt find answers to that?

**Comment by meltedspork** [comment]
screen shot 2018-02-06 at 6 37 57 pm
screen shot 2018-02-06 at 6 37 12 pm
screen shot 2018-02-06 at 6 37 00 pm
screen shot 2018-02-06 at 6 36 52 pm



@kratsg can you explain what you mean by this? We are able to run it via docker. See screenshots if it helps clears up? [](url)

File: src/index.js (hunk 1 above)
Comment (kratsg) [reply src/index.js]: Probably use this middleware: https://github.com/Tabcorp/restify-cors-middleware

**Comment by kratsg** [comment]
I mean being able to test the spec against the existing api.

**Comment by meltedspork** [comment]
Depends on what your definition by "spec" but this is how we test specs with swagger.

screen shot 2018-02-06 at 6 37 12 pm

**Comment by kratsg** [comment]
It's not automated? So we can't catch issues with the documentation and not matching the right JSON schema for outputs? This is something I definitely want.

File: src/index.js (hunk 1 above)
Comment (meltedspork) [reply src/index.js]: Ahh, ill take a look at that npm. I was hoping that there was more builtin core. But this probably will work.

Comment (kratsg) [reply src/index.js]: I believe most plugins got migrated. I can check the core code later to see if there's a specific plugin for it.

Comment (kratsg) [reply src/index.js]: http://restify.com/docs/4to5/#cors -- looks like the plugin is right.

**Comment by meltedspork** [comment]
it can be - https://assertible.com - https://assertible.com/blog/testing-an-api-using-swagger i just signed up seems lot of limit amount of using it but i think we can work with it?

I just invited you from there.

File: src/index.js (hunk 1 above)
Comment (meltedspork) [reply src/index.js]: 👍 works for me

**Comment by kratsg** [comment]
Why does the CORS thing use `API-Token` in the header and not the Bearer that we normally expect, or am I not understanding CORS?

**Comment by meltedspork** [comment]
Whoops that part wasn't supposed be committed in.  Ill remove that

**Comment by meltedspork** [comment]
I think this is good now. with your apiDoc and Dredd I think we will have for that part of automation set for both. Then we can add unit tests as needed. Along improvement with swagger opens-api 3.0 as well

**Comment by kratsg** [comment]
Yeah, I think we're going with OpenAPI (not entirely sold on swagger itself...).

**Review by kratsg — APPROVED** [review]

**Comment by meltedspork** [comment]
Yeah if we are not still sold on swagger, we can always remove swagger stuff 👍

**Pull request merged**